aws_lambda_powertools/utilities/parser/functions.py

```diff
@@ -82,3 +82,28 @@ def _parse_and_validate_event(data: dict[str, Any] | Any, adapter: TypeAdapter):
             data = json.loads(data)
 
     return adapter.validate_python(data)
+
+
+def _validate_source_ip(value):
+    """
+    Handle sourceIp that may come with port (e.g., "10.1.15.242:39870")
+    in certain network configurations like Cloudflare + CloudFront + API Gateway.
+    Validates the IP part while preserving the original format.
+    See: https://github.com/aws-powertools/powertools-lambda-python/issues/7288
+    """
+
+    if value == "test-invoke-source-ip":
+        return value
+
+    try:
+        # The value is always an instance of str before Pydantic validation occurs.
+        # So the first thing to do is try to convert it.
+        IPvAnyNetwork(value)
+    except ValueError:
+        try:
+            ip_part = value.split(":")[0]
```

Comment:
Has "IPv6-with-port" been taken into account?

Reply:
As far as I understand, the IPvAnyAddress model we're using supports both v4 and v6 - [source](https://docs.pydantic.dev/latest/api/networks/#pydantic.networks.IPvAnyAddress)

Reply:
But I think the way we split the port might actually not work with IPv6 which contains multiple `:`. To make it work we should've taken everything but the last item of the split list instead of just the first. cc @leandrodamascena

Reply:
And probably with `.strip("[]")` as well, if `IPvAnyNetwork` doesn't expected bracketed IPv6 addresses.